### PR TITLE
Deprecate soft_deletes.enabled setting

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -48,7 +48,9 @@ None
 Deprecations
 ============
 
-None
+- The setting ``soft_deletes.enabled`` has been marked as deprecated and will be
```
Comment (norosa) [inline docs/appendices/release-notes/unreleased.rst:51]: nit: "be" should be on the next line (79 chars max)

File: server/src/test/java/org/elasticsearch/index/engine/NoOpEngineTests.java [hunk 2]
```diff
@@ -54,6 +54,11 @@
 public class NoOpEngineTests extends EngineTestCase {
     private static final IndexSettings INDEX_SETTINGS = IndexSettingsModule.newIndexSettings("index", Settings.EMPTY);
 
+    @Override
+    protected boolean enableWarningsCheck() {
```
Comment (mkleen) [inline server/src/test/java/org/elasticsearch/index/engine/NoOpEngineTests.java:58]: Deprecation warning should not break this test.

**Review by seut — APPROVED** [review]

File: docs/appendices/release-notes/unreleased.rst [hunk 3]
```diff
@@ -48,7 +48,9 @@ None
 Deprecations
 ============
 
-None
+- The setting ``soft_deletes.enabled`` has been marked as deprecated and will
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:51]:
```suggestion
- The table setting :ref:`sql-create-table-soft-deletes-enabled` has been marked as deprecated and will
```

**Comment by seut** [comment]
@mergifyio rebase

**Pull request merged**